File: dotnet/Solid/Solid/LSP/Example2/Domain/PaypalPayment.cs
using Solid.LSP.Example2.External;

namespace Solid.LSP.Example2.Domain
{
    public class PayPalPayment : Payment
    {
        public string Username { private get; set; }
        public string Password { private get; set; }

        public override string Refund(decimal amount, string transactionId)
        {
            PayPalWebService payPalWebService = new PayPalWebService();
            string token = payPalWebService.GetTransactionToken(Username, Password);
            string response = payPalWebService.MakeRefund(amount, transactionId, token);
            return response;
        }
    }
}

File: java/liskov-plus-start/src/main/java/no/bouvet/solid/lsp/domain/PayPalPayment.java
package no.bouvet.solid.lsp.domain;

import java.math.BigDecimal;

import no.bouvet.solid.lsp.external.PayPalWebService;

public class PayPalPayment extends Payment {
    private String username;
    private String password;

    public void setUsername(String username) {
        this.username = username;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    @Override
    public String refund(BigDecimal amount, String transactionId) {
        PayPalWebService payPalWebService = new PayPalWebService();
        String token = payPalWebService.getTransactionToken(username, password);
        String response = payPalWebService.makeRefund(amount, transactionId, token);
        return response;
    }

}
